### DISCUSSION TITLE
Transaction manager on kubernetes

### DISCUSSION THREAD

**Discussion by jjansenvr** [frankframework/frankframework#7098] [General]
What are the best practices on using transaction manager in a K8s environment

Options for storing TX logfiles
- Narayana using a local folder (losing it on every reboot)
- Narayana using a local folder on a statefullset
- Narayana using a Database log

**Comment by nielsm5** [comment]
You've pretty covered all 3 options!
Option 1 is configuration-less. This happens out-of-the-box.
Option 2 requires some k8s configuration..
Options 3 has been documented here:
https://frank-manual.readthedocs.io/en/latest/advancedDevelopment/narayana/narayana.html

**Reply by Laurens-makel** [reply]
Correct me if I''m wrong here please, but in order to implement option 3, you'd need a statefulset as well (or other type of k8s configuration) in order to provide the UID mentioned in the documentation.

This UID needs to be a constant value that is bound to a specific node, if you'd use a `kind: Deployment`, you'd have a random hostname every reboot thus having a random UID everytime. Using a statefulset you'd get a constant UID value for each unique node in the cluster (if node 2 goes down, it will come up as node 2 instead of a random new node name when using Deployment), which is required to pickup transaction logs in case of a ungraceful shutdown of a specific instance.
